apis/fluentbit/v1alpha2/plugins/output/stackdriver_types.go

```diff
@@ -0,0 +1,143 @@
+package output
+
+import (
+	"fmt"
+	"github.com/fluent/fluent-operator/apis/fluentbit/v1alpha2/plugins"
+	"github.com/fluent/fluent-operator/apis/fluentbit/v1alpha2/plugins/params"
+	"github.com/fluent/fluent-operator/pkg/utils"
+)
+
+// +kubebuilder:object:generate:=true
+
+// Stackdriver is the Stackdriver output plugin, allows you to ingest your records into GCP Stackdriver.
+type Stackdriver struct {
+	// Path to GCP Credentials JSON file
+	GoogleServiceCredentials string `json:"googleServiceCredentials,omitempty"`
+	// Email associated with the service
+	ServiceAccountEmail *plugins.Secret `json:"serviceAccountEmail,omitempty"`
+	// Private Key associated with the service
+	ServiceAccountSecret *plugins.Secret `json:"serviceAccountSecret,omitempty"`
+	// Metadata Server Prefix
+	MetadataServer string `json:"metadataServer,omitempty"`
+	// GCP/AWS region to store data. Required if Resource is generic_node or generic_task
+	Location string `json:"location,omitempty"`
+	// Namespace identifier. Required if Resource is generic_node or generic_task
+	Namespace string `json:"namespace,omitempty"`
+	// Node identifier within the namespace. Required if Resource is generic_node or generic_task
+	NodeID string `json:"nodeID,omitempty"`
+	// Identifier for a grouping of tasks. Required if Resource is generic_task
+	Job string `json:"job,omitempty"`
+	// Identifier for a task within a namespace. Required if Resource is generic_task
+	TaskID string `json:"taskID,omitempty"`
+	// The GCP Project that should receive the logs
+	ExportToProjectID string `json:"exportToProjectID,omitempty"`
+	// Set resource types of data
+	Resource string `json:"resource,omitempty"`
+	// Name of the cluster that the pod is running in. Required if Resource is k8s_container, k8s_node, or k8s_pod
+	K8sClusterName string `json:"k8sClusterName,omitempty"`
+	// Location of the cluster that contains the pods/nodes. Required if Resource is k8s_container, k8s_node, or k8s_pod
+	K8sClusterLocation string `json:"k8sClusterLocation,omitempty"`
+	// Used by Stackdriver to find related labels and extract them to LogEntry Labels
+	LabelsKey string `json:"labelsKey,omitempty"`
+	// Optional list of comma separated of strings for key/value pairs
+	Labels []string `json:"labels,omitempty"`
```

Comment:
This can also be left unchanged, but I think the data structure of this `kv` structure can be used with a map. like this
https://github.com/fluent/fluent-operator/blob/ed10b753faa66765930e958dbbd37a9fb4eeed65/apis/fluentbit/v1alpha2/plugins/filter/modify_types.go#L63

Reply:
Ah, good to know - thanks!